lib/net.c

```diff
@@ -1517,18 +1517,16 @@ start_port(uint8_t port_id, uint8_t *pnum_succ_ports,
 	 * are activated on a timer, so this can take some time.
 	 */
 	do {
-		rte_eth_link_get(port_id, &link);
-
+		ret = rte_eth_link_get(port_id, &link);
 		/* Link is up. */
-		if (link.link_status)
+		if (ret == 0 && link.link_status)
 			break;
 
-		G_LOG(ERR, "net: querying port %hhu, and link is down\n",
-			port_id);
+		G_LOG(ERR, "net: querying port %hhu failed with err - %s, and link is down\n",
+			port_id, rte_strerror(-ret));
```

Comment:
We need a log entry for a possible error coming from `rte_eth_link_get()`. The log entry above is for when the link is down.

If we deal with errors immediately after the return of `rte_eth_link_get()`, the changes after line `ret = rte_eth_link_get(port_id, &link);` are no longer needed.